mlflow/types/responses.py

```diff
@@ -0,0 +1,83 @@
+import json
+import warnings
+from typing import Any, Optional, Union
+
+from pydantic import ConfigDict
+
+from mlflow.types.agent import ChatContext
+from mlflow.types.chat import BaseModel
+from mlflow.types.responses_helpers import (
+    BaseRequestPayload,
+    Message,
+    OutputItem,
+    Response,
+    ResponseCompletedEvent,
+    ResponseErrorEvent,
+    ResponseOutputItemDoneEvent,
+    ResponseTextAnnotationDeltaEvent,
+    ResponseTextDeltaEvent,
+    Tools,
+)
+from mlflow.types.schema import Schema
+from mlflow.types.type_hints import _infer_schema_from_type_hint
+from mlflow.utils.pydantic_utils import model_validator
+
+
```

Comment:
Side note for @serena-ruan : lots of Union/Any are used, we probably should remove the warnings for using them (or silent them here at least)